mlflow/tracking/utils.py

```diff
@@ -0,0 +1,112 @@
+from __future__ import print_function
+
+import os
+
+from six.moves import urllib
+
+from mlflow.store.file_store import FileStore
+from mlflow.store.rest_store import RestStore, DatabricksStore
+from mlflow.store.artifact_repo import ArtifactRepository
+from mlflow.utils import env, rest_utils
+
+
+_TRACKING_URI_ENV_VAR = "MLFLOW_TRACKING_URI"
+_LOCAL_FS_URI_PREFIX = "file:///"
+_REMOTE_URI_PREFIX = "http://"
+
+_tracking_uri = None
+
+
+def set_tracking_uri(uri):
```

Comment:
It seems that this and `get_tracking_uri` are public functions, so do we want to move them or keep them in `tracking`? The rest of the stuff in this module is private. I guess you can set them through `mlflow` directly so it may be OK to move them, but it might also be better to just keep this module private.

Reply:
The dependency structure is a bit weird. `_get_store()` has a lot of helper utilities to it (which is why I wanted to move it to its own py file), and it relies on `_tracking_uri` which is set by `set_tracking_uri`. `fluent` and `service` depend on `utils`, so I can't move `_tracking_uri`.

I could move the set/get methods to mutate a global specified in `utils.py`, but I'm not sure that's cleaner.